File: 6th sem/C++/A1/q8.cc
#include <iostream>
using namespace std;

double pow(double m, int n = 2) {
   double res = 1;
   int i;
   for (i = 0; i < n; i++)
      res *= m;
   return res;
}

int main() {
   double z;
   int n;
   cout << "Enter a number: ";
   cin >> z;
   cout << "Enter power: ";
   cin >> n;

   cout << "Without default parameter" << pow(z, n) << endl;
   cout << "With default parameter" << pow(z) << endl;

   return 0;
}

File: 6th sem/SSWD/A1/q8.mjs
import http from "http"
import url from "url"

http.createServer((req, res) => {
   res.writeHead(200, { "content-type": "text/plain" })
   const qS = url.parse(req.url, true).query
   const n = qS.n
   if (!isNaN(n)) {
      let sum = 0;
      const digits = n.toString().split("").map(Number);
      const power = digits.length;

      for (let digit of digits) {
         sum += Math.pow(digit, power);
      }

      res.end(`${n} is${n == sum ? "" : " not"} armstrong`)
   } else
      res.end("Provide a number as a query parameter: ?n=45")
})
.listen(8125, "127.0.0.2", () => console.log('http://127.0.0.2:8125'))